controllers/swiftstorage_controller.go

```diff
@@ -217,6 +220,53 @@ func (r *SwiftStorageReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		if err != nil {
 			return ctrl.Result{}, err
 		}
+
+		// When the cluster is attached to an external network, create DNS record for every
+		// cluster member so it can be resolved from outside cluster (edpm nodes)
+		podList, err := pod.GetPodListWithLabel(ctx, helper, instance.Namespace, serviceLabels)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		for _, swiftPod := range podList.Items {
+			dnsIP := ""
+			if len(instance.Spec.NetworkAttachments) > 0 {
+				// Currently only IPv4 is supported
+				dnsIP, err = getPodIPv4InNetwork(swiftPod, instance.Namespace, "storage")
+				if err != nil {
+					return ctrl.Result{}, err
+				}
+			}
+
+			if len(dnsIP) == 0 {
+				// If this is reached it means that no IP was found in the network
+				// or no networkAttachment exists. Try to use podIP if possible
+				if len(swiftPod.Status.PodIP) > 0 {
+					dnsIP = swiftPod.Status.PodIP
+				}
+			}
+
+			if len(dnsIP) == 0 {
+				return ctrl.Result{}, err
```

Comment:
`err` could still be `nil` here.  Is the `return` intended to convey an error, or is it just to kick-out of the reconcile?  If its the former, I would suggest explicitly returning a newly-created error.

Reply:
Oh right, this should raise an actual error and not be nil at this point. Just pushed updated commit. Thx!